src/traces/bar/plot.js

```diff
@@ -647,10 +647,14 @@ function calcTexttemplate(fullLayout, cd, index, xa, ya) {
     }
 
     function formatLabel(u) {
+        if(pAxis.type === 'log') u = pAxis.c2r(u);
+
         return tickText(pAxis, u, true).text;
     }
 
     function formatNumber(v) {
+        if(vAxis.type === 'log') v = vAxis.c2r(v);
```

Comment:
Nonblocking, but `formatLabel` and `formatNumber` are nearly the same function - just `pAxis` vs `vAxis` and the `+` in the `tickText` call (any reason that's in one but not the other?) - so should be able to be collapsed.

Reply:
Actually, we may be able to simplify all of this, and get rid of the `type === 'log'` checks entirely, by using `c2l`

https://github.com/plotly/plotly.js/blob/8cccacc89e3f4cbccc2c02e366efebd1dfe44651/src/plots/cartesian/set_convert.js#L243

And while I'm here, this is `ensureNumber`:

https://github.com/plotly/plotly.js/blob/8cccacc89e3f4cbccc2c02e366efebd1dfe44651/src/lib/index.js#L166-L171

Any reason the last line isn't just `return v;`? This is a pretty hot function. After already filtering out non-numeric values and converting others to `Number` explicitly, why would we have to check `isNumeric` again?

Reply:
> Actually, we may be able to simplify all of this, and get rid of the `type === 'log'` checks entirely, by using `c2l`
> 
> https://github.com/plotly/plotly.js/blob/8cccacc89e3f4cbccc2c02e366efebd1dfe44651/src/plots/cartesian/set_convert.js#L243
> 

Good call. Applied in https://github.com/plotly/plotly.js/pull/5622/commits/ca962bc0a6adb687416f332ecbf769d2f215ff9d.

Reply:
The `ensureNumber` comment is tracked in #5634.